### DISCUSSION TITLE
Real-time beatmap downloading

### DISCUSSION THREAD

**Discussion by TheBrunoRM** [ppy/osu#27038] [Ideas] [answered]
I thought about this really cool but crazy idea. Imagine being able to play a beatmap while it downloads. This is probably very out of scope, but I wanted to share it since it seems nobody has talked about this before.

I was thinking it could work like this: when downloading a beatmap, it would download the beatmap .osu file(s) (which are very lightweight most of the time), and then it would download the background and the audio in a stream-like way, similar to any stream service, using buffering and such. This would allow you to play a beatmap while it is still downloading the audio while playing it.

In a practical way, this is not that useful, as it would barely reduce downloading times and it would just introduce complexity.
I wanted to share this idea here nontheless, since I think it would be super cool, even if it is super out of scope.

**Comment by peppy** [comment, marked as answer]
This was originally something I wanted to see happen. The way we've structured things will allow it to be implemented quite easily, but there's no timeframe on when (it's one of those nice-to-have but also low priority features).
